dawn/src/dawn/AST/ASTStringifier.cpp

```diff
@@ -229,8 +229,13 @@ class StringVisitor : public ASTVisitor {
   }
 
   void visit(const std::shared_ptr<FieldAccessExpr>& expr) override {
+    auto offset = expr->getOffset();
+    auto const& hOffset = ast::offset_cast<CartesianOffset const&>(offset.horizontalOffset());
```

Comment:
Later there will be a switch between Cartesian and unstructured?

Reply:
In a follow-up PR I moved hOffset into the else case.  The case where "FieldAccessExpr" has arguments (not sure what that means), is not supported in unstructured anyway. Maybe, toString will be able to handle this at that point, but there are missing pieces, like the argumentMap and argumentOffset.